<a href="https://colab.research.google.com/github/veera37/data-science-assignments/blob/main/basic_statistics_level_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**SET 1**  **Descriptive+statistics+Probability**

Q2

In [4]:
#problem on binomial distribution 
#here n = 5 , p = to one in 200 ie 0.005
from scipy.stats import binom
bi = binom(5,0.005)
bi

In [5]:
# asked atleast one out of 5 reaches wrong number
bi.cdf(1)
1-bi.cdf(1)

0.0002475093624999536

In [ ]:
##so, there is 0.00024% of chance for one ot of 5 calls getting diverted. 



--------------------------------------------------------------------------------



***Set2:***
 **Normal distribution, Functions of Random Variables**

Q1

-> starts work after 10 minutes

-> coustmer returns after one hour. so the time left is 50 minutes

->mean(mu) is 45

->standrad deviation is (sigma) is 8 minutes


In [6]:
from scipy.stats import norm
nd = norm(45,8)
nd.cdf(50)
p = 1 - nd.cdf(50)
p

0.26598552904870054

Q2.a

In [ ]:
from scipy.stats import norm
nd = norm(38,6)
p1 = nd.cdf(44)
p2 = nd.cdf(38)
print("p(38>=44) is ",p1 - p2)#the number of employes between 38 and 44
print('older than 44 are ',1 - p1)#the employes older than 44

p(38>=44) is  0.3413447460685429
older than 44 are  0.15865525393145707


p(38>=44) is  0.3413447460685429
older than 44 are  0.15865525393145707

there are more employes between 38 and 44 age group.

Q2.b

In [ ]:
pp = nd.cdf(30)
p3

9.121121972586788

there are 9% of employes under 30 age group.. whch is equal to 36 employes out of 400.


Q4

In [ ]:
from scipy import stats
from scipy.stats import norm

In [ ]:
# p =99 mean = 100, sd=20
stats.norm.interval(0.99,20,4.47)

(88.48604301313641, 111.51395698686359)

Q5.a


In [ ]:
from scipy import stats
from scipy.stats import norm
stats.norm.interval(0.95,12,5)

(2.2001800772997306, 21.79981992270027)

z value for 5% percentail is  -1.645

In [ ]:
# percentile 
12 + (-1.645*5) 

3.7750000000000004



SET 3

confidence interval

Q 8

In [18]:
from scipy.stats import norm 
norm.ppf(0.95)

1.6448536269514722

N= z^2 (p * 1-p ) / E^2

Z – z score for 95 %

P is not given , so default value is 0.5

E is margin of error = 0.04

In [ ]:
(1.959963984540054*1.959963984540054*0.5*0.5)/(0.04*0.04) 

600.2279407334571

Q 9

In [ ]:
norm.ppf(0.99)

2.3263478740408408

In [ ]:
(2.3263478740408408*2.3263478740408408*0.5*0.5)/(0.04*0.04) 

845.6085048522405

**set 4**

**Q 3** 


given that:
-> no of samples are 100 

-> population mean is 50 

-> sd of population is 40

here we need find the probality of no investigation ie 1 - p(45>=55)

to find that we need sd of sample . for that           

In [36]:
#sd of sample = sd of pop/ sq(no of samples)
import math
from scipy import stats
from scipy.stats import norm
sd_sam =40/(math.sqrt(100))
p4 = stats.norm.cdf(55,50,sd_sam) - stats.norm.cdf(45,50,sd_sam)
print("probability of no investigation is ",1 - p4)



probability of no investigation is  0.2112995473337107
